tests/Doctrine/Tests/ODM/PHPCR/Functional/Tools/Helper/TranslationConverterTest.php

```diff
@@ -0,0 +1,445 @@
+            ),
+            'old property was not removed'
+        );
+
+        $commentDoc = $this->dm->find(null, '/functional/convert');
+        $this->assertInstanceof($class, $commentDoc);
+        $this->assertEquals('Lorem ipsum...', $commentDoc->getText());
+
+        $this->dm->clear();
+
+        $commentDoc = $this->dm->find(null, '/functional/convert');
+        $this->assertInstanceof($class, $commentDoc);
+        $this->assertEquals('Lorem ipsum...', $commentDoc->getText());
+    }
+
+    public function testUntranslateAttribute()
+    {
+        $class = 'Doctrine\Tests\Models\Blog\Comment';
+        $field = 'title';
+        $comment = $this->node->addNode('convert');
+        $comment->setProperty(
+            AttributeTranslationStrategyTest::propertyNameForLocale('en', $field),
+            'Lorem ipsum...'
+        );
+        $comment->setProperty('phpcr:class', $class);
+        $this->session->save();
+
+        $this->assertFalse($this->converter->convert($class, 'attribute', array($field)));
+
+        $this->session->save();
+        $this->assertTrue(
+            $comment->hasProperty($field),
+            'new property was not created'
+        );
+
+        $this->assertFalse(
+            $comment->hasProperty(
+                AttributeTranslationStrategyTest::propertyNameForLocale('en', $field)
+            ),
+            'old property was not removed'
+        );
+
+        $commentDoc = $this->dm->find(null, '/functional/convert');
+        $this->assertInstanceof($class, $commentDoc);
+        $this->assertEquals('Lorem ipsum...', $commentDoc->title);
+
+        $this->dm->clear();
+
+        $commentDoc = $this->dm->find(null, '/functional/convert');
+        $this->assertInstanceof($class, $commentDoc);
+        $this->assertEquals('Lorem ipsum...', $commentDoc->title);
+    }
+
+    /**
+     * Just make one field of a document untranslated again
+     */
+    public function testPartialUntranslateAttribute()
+    {
+        $article = new Article();
+        $article->id = '/functional/convert';
+        $article->topic = 'Some interesting subject';
+        $article->setText('Lorem ipsum...');
+        $this->dm->persist($article);
+        $this->dm->flush();
+        $this->dm->clear();
+
+        $class = 'Doctrine\Tests\Models\Translation\Article';
+        $field = 'author';
+        $comment = $this->node->getNode('convert');
+        $comment->setProperty(
+            AttributeTranslationStrategyTest::propertyNameForLocale('en', $field),
+            'Move to untranslated'
+        );
+        $this->session->save();
+
+        $this->assertFalse($this->converter->convert($class, null, array($field)));
+        $this->session->save();
+        $this->dm->clear();
+
+        $this->dm = $this->createDocumentManager();
+        $this->dm->setLocaleChooserStrategy(new LocaleChooser($this->localePrefs, 'en'));
```

Comment:
at this point the metadata is pretty much f*ked up. we might try to restore it. but the doc should tell to not even try to use the dm anymore after a migration.